src/Components/Toolbar/SaveCharacter/SaveAlerts/displayAlert.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { useAppSelector } from 'features/reduxHooks';
+import { Alert } from 'react-bootstrap';
```

Comment:
```suggestion
import { Alert as BaseAlert } from 'react-bootstrap';

export const DisplayAlert: React.FC<{ alertType: string }> = ({ alertType }) => {
  const characterName = useAppSelector((state) => state.character.name);
  const nameText = characterName;
  const [show, setShow] = useState(true);

  const alerts = {
    success: {
      type: 'success',
      msg: `${nameText} has been succesfully saved.`
    },
    'no-attributes': {
      type: 'no-attributes',
      msg: 'You must select at least one Advantage or Disadvantage.'
    },
    duplicate: {
      type: 'duplicate',
      msg: 'You already have a character with this name. Delete original character or change name.'
    },
    'no-name': {
      type: 'no-name',
      msg: 'You must type a name for your character and press enter in order to save.'
    },
    'save-error': {
      type: 'save-error',
      msg: 'Failed to save to database, please try again later.'
    }
  };

  const alert = alerts[alertType];

  const Alert = (props: { variant: string; text: string }) => (
    <div className="save-alert">
      <BaseAlert variant={props.variant} onClose={() => setShow(false)} dismissible>
        <BaseAlert.Heading>{props.text}</BaseAlert.Heading>
        <div className="d-flex justify-content-end"></div>
      </BaseAlert>
    </div>
  );

  return show ? <Alert variant={alert.type} text={alert.msg} /> : null;
};

```

dunno just an alternative, bootstrap is typed horribly btw. maybe its better in future versions. You've also got a conditional where DispalyAlert is called to show/hide it. id remove that because this already takes the show prop. there's a bunch of ways you could structure this, you're extreme prettier settings forced each to be a multi line haha. Just having the text grouped In an object makes it a bit easier to change or add alerts in the future.

Reply:
Youre suggested changes are great, so much cleaner and readable.
I have incorperated them and used a similar method in my portfolio.